tests/stats/stats-calculator.php

```diff
@@ -51,43 +51,62 @@ public function test_calculates_stats_for_event() {
-		$this->stats_factory->create( $event1_id, $user3_id, 22, 'request_changes', 'bb' );
+		$translation_set_2 = $this->factory->translation_set->create_with_project_and_locale();
+		$original_21       = $this->create_original_and_translation( $translation_set_2 );
+		$original_22       = $this->create_original_and_translation( $translation_set_2 );
+		$this->stats_factory->create( $event1_id, $user1_id, $original_21->id, 'create', $translation_set_2->locale );
+		$this->stats_factory->create( $event1_id, $user2_id, $original_22->id, 'create', $translation_set_2->locale );
+		$this->stats_factory->create( $event1_id, $user3_id, $original_21->id, 'approve', $translation_set_2->locale );
+		$this->stats_factory->create( $event1_id, $user3_id, $original_22->id, 'request_changes', $translation_set_2->locale );
 
 		// For event2, which should not be included in the stats.
-		$this->stats_factory->create( $event2_id, $user1_id, 31, 'create' );
-		$this->stats_factory->create( $event2_id, $user1_id, 32, 'create' );
-		$this->stats_factory->create( $event2_id, $user2_id, 31, 'approve' );
-		$this->stats_factory->create( $event2_id, $user2_id, 32, 'reject' );
+
+		$this->stats_factory->create( $event2_id, $user1_id, 31, 'create', $translation_set_1->locale );
+		$this->stats_factory->create( $event2_id, $user1_id, 32, 'create', $translation_set_1->locale );
+		$this->stats_factory->create( $event2_id, $user2_id, 31, 'approve', $translation_set_1->locale );
+		$this->stats_factory->create( $event2_id, $user2_id, 32, 'reject', $translation_set_1->locale );
 
 		$event1 = get_post( $event1_id );
 		$stats  = $this->calculator->for_event( $event1->ID );
-
 		$this->assertCount( 2, $stats->rows() );
 
-		// Locale aa.
-		$this->assertEquals( 3, $stats->rows()['aa']->created );
-		$this->assertEquals( 3, $stats->rows()['aa']->reviewed );
-		$this->assertEquals( 3, $stats->rows()['aa']->users );
+		// $translation_set_1 Locale.
+		$this->assertEquals( 3, $stats->rows()[ $translation_set_1->locale ]->created );
+		$this->assertEquals( 3, $stats->rows()[ $translation_set_1->locale ]->reviewed );
+		$this->assertEquals( 3, $stats->rows()[ $translation_set_1->locale ]->users );
 
-		// Locale bb.
-		$this->assertEquals( 2, $stats->rows()['bb']->created );
-		$this->assertEquals( 2, $stats->rows()['bb']->reviewed );
-		$this->assertEquals( 3, $stats->rows()['bb']->users );
+		// $translation_set_2 Locale.
+		$this->assertEquals( 2, $stats->rows()[ $translation_set_2->locale ]->created );
+		$this->assertEquals( 2, $stats->rows()[ $translation_set_2->locale ]->reviewed );
+		$this->assertEquals( 3, $stats->rows()[ $translation_set_2->locale ]->users );
 
 		// Totals.
 		$this->assertEquals( 5, $stats->totals()->created );
 		$this->assertEquals( 5, $stats->totals()->reviewed );
 		$this->assertEquals( 3, $stats->totals()->users );
 	}
+
+	private function create_original_and_translation( $translation_set, $status = 'current' ) {
```

Comment:
Instead of adding this method, we could add an optional `$translation_set` and `$status` argument to the existing `create()` function in `tests/lib/translation-factory.php`. Alternatively, we could move this function as is to `tests/lib/translation-factory.php`.